File: DataStructuresInJava/src/main/java/implementations/linkedLists/LinkedListExample.java
package implementations.linkedLists;

import java.util.Iterator;
import java.util.LinkedList;
import java.util.ListIterator;

public class LinkedListExample {
    public static void LinkedListBase()
    {
        //Linked list instantiation
        LinkedList<String> pages = new LinkedList<>();

        //Adding and removing nodes
        pages.add("test1");
        pages.addFirst("test0");
        pages.addLast("test2");
        pages.add("test3");
        Boolean isContains = pages.contains("test1");

        //Get values
        String search1 = pages.getFirst();
        String search2 = pages.get(2);

        //Get count
        int count = pages.size();

        //Node iterator
        Iterator<String> iterator = pages.iterator();
        while (iterator.hasNext()){
            System.out.print("-->" + iterator.next());
        }

        ListIterator<String> li = pages.listIterator(0);
        while (li.hasNext()) {
            //your logic goes here
            //if you need to go to the previous place
            if (li.hasPrevious()) {
                li.previous();
                //further logic here...
            }
        }

        //Remove
        pages.remove("test0");
        pages.remove(0);
        pages.removeFirst();
        pages.removeLast();
    }

}

File: DataStructuresInCSharp/DataStructuresInCSharp/Implementations/2_LinkedLists/LinkedListExample.cs
using DataStructuresInCSharp.Implementations._CommonSteps;
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;

namespace DataStructuresInCSharp.Implementations._2_LinkedLists
{
    public class LinkedListExample
    {
        public static void LinkedListBase()
        {
            //Linked list instantiation
            LinkedList<string> pages = new LinkedList<string>();

            //Adding and removing nodes
            LinkedListNode<string> node1 = pages.AddLast("test1");
            LinkedListNode<string> node2 = pages.AddFirst("test2");
            LinkedListNode<string> node3 = pages.AddBefore(node1, "test3");
            LinkedListNode<string> node4 = pages.AddAfter(node3, "test4");

            LinkedListNode<string> currentEl = pages.First;
            Console.Write(currentEl.Value);
            while (currentEl.Next != null)
            {
                currentEl = currentEl.Next;
                Console.Write("-->" + currentEl.Value);
            }

            bool isContains = pages.Contains("test1");
            pages.RemoveFirst();
            pages.RemoveLast();
            pages.Remove("test3");

            //Operations with node
            LinkedListNode<string> searchNode = pages.Find("test4");
            string value = searchNode.Value;
            LinkedListNode<string> prevNode = searchNode.Next;
            LinkedListNode<string> nextNode = searchNode.Previous;

            //Instantiate and fill
            LinkedList<int> pages2 = new LinkedList<int>(new int[] { 2, 2, 5, 2 });
        }

        public static LinkedList<string> RealWorldExample1(IList<string> pagesToNavigate)
        {
            //Init
            var highScorePages = new LinkedList<string>();
            string baseUrl = "https://football.ua/";
            IWebDriver webDriver = WebDriverInitialization.Instance.Driver;
            

            //Act
            //Navigate pages and receive goals
            pagesToNavigate.ToList().ForEach(page => 
                { 
                    webDriver.Navigate().GoToUrl(baseUrl + page);
                    highScorePages.AddLast(baseUrl + page);
                }
            );

            Dictionary<string, int> scorePerPage = PagesDbEmulator.GetScorePerPage();
            LinkedListNode<string> node = highScorePages.First;

            while (!node.Equals(null))
            {
                int score = scorePerPage[node.Value];
                if (score < 11)
                    highScorePages.Remove(node);
                node = node.Next;
            }
            return highScorePages;
        }
    }
}
